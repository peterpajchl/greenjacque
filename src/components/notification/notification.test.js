import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';
import Notification from './notification';

Enzyme.configure({ adapter: new Adapter() });

describe('<Notification /> component', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Notification />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders an empty message when not set', () => {
    const wrapper = shallow(<Notification />);
    expect(wrapper.find('p').length).toEqual(1);
    expect(wrapper.find('p').first().text()).toEqual('');
  });

  it('renders a single message when set', () => {
    const wrapper = shallow(<Notification msg="Hello" />);
    expect(wrapper.find('p').length).toEqual(1);
    expect(wrapper.find('p').first().text()).toEqual('Hello');
  });

  it('has correct css classes when displayed', () => {
    const wrapper = shallow(<Notification msg="Hello" hasMsg={true} />);
    expect(wrapper.hasClass('notification notification--info notification--visible slide-down')).toBeTruthy();
  });

});