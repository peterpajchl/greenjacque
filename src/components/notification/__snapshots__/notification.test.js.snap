// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> component should match the snapshot 1`] = `"<div class=\\"notification notification--info \\"><p class=\\"notification__text\\"></p></div>"`;
